components/validate.js

```diff
@@ -0,0 +1,77 @@
+class Validator {
+    validateSet(set) {
+        var durationCheck = false;
+        var notesCheck = false;
+        var repetitionCheck = false;
+        var weightCheck = false;
+        if (set != undefined && set != null) {
+            if (set.duration > 0) {
+                durationCheck == true;
+            }
+            if (set.notes.length < 100) {
+                notesCheck == true;
+            }
+            if (set.repetitions > 0) {
+                repetitionCheck == true;
+            }
+            if (Number.isInteger(set.weight)) {
+                weightCheck == true;
+            }
+        }
+
+        if (durationCheck && notesCheck && repetitionCheck && weightCheck) {
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    validateExercise(exercise) {
+        var nameCheck = false;
+        var descriptionCheck = false;
+        var setCheck = false;
+        if (exercise != undefined && exercise != null && exercise.id != null) {
+            if (exercise.name.length > 0 && exercise.name.length < 50) {
+                nameCheck == true;
+            }
+            if (exercise.description.length < 200) {
```

Comment:
Description is optional, so make sure it doesn't fail if it's not present.

Reply:
I think exercise description comes from a database depending on exercise type, unless it is a custom exercise. So yes, should not fail if not present.